src/main/java/seedu/address/logic/commands/BestCommand.java

```diff
@@ -0,0 +1,189 @@
+    public static final String MESSAGE_SUCCESS = "Here are the best %s interviewees based on %s";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds best N candidates from the list.\n"
+            + "Parameters: "
+            + " NUMBER_OF_INTERVIEWEES [-a ATTRIBUTE_PREFIX] [-m METRIC_PREFIX]\n"
+            + "e.g. best 3 -a lea";
+
+    private final String numberOfInterviewees;
+    private final String paramPrefix;
+    private final BestParameter paramType;
+
+    public BestCommand(String numberOfInterviewees, BestParameter paramType) {
+        this.numberOfInterviewees = numberOfInterviewees;
+        this.paramPrefix = "";
+        this.paramType = paramType;
+    }
+
+    public BestCommand(String numberOfInterviewees, String paramPrefix, BestParameter paramType) {
+        this.numberOfInterviewees = numberOfInterviewees;
+        this.paramPrefix = paramPrefix;
+        this.paramType = paramType;
+    }
+
+    /**
+     * Executes the best command.
+     * @param model {@code Model} which the command should operate on.
+     * @return The corresponding CommandResult instance.
+     * @throws CommandException If there is an invalid parameter entered by the client.
+     */
+    public CommandResult execute(Model model) throws CommandException {
+        if (!model.isFinalisedInterviewProperties()) {
+            throw new CommandException(MESSAGE_HAS_NOT_FINALIZED);
+        }
+        int size = parseNumberOfInterviewees(numberOfInterviewees);
+        Comparator<Interviewee> comparator;
+        try {
+            switch (paramType) {
+            case METRIC:
+                comparator = getComparatorByMetric(model.getMetricList(), paramPrefix);
+                break;
+            case ATTRIBUTE:
+                comparator = getComparatorByAttribute(model.getAttributeList(), paramPrefix);
+                break;
+            default:
+                comparator = getOverallComparator(model.getAttributeList());
+            }
+
+            ObservableList<Interviewee> observableInterviewees = model.getIntervieweeList().getObservableList();
+            ObservableList<Interviewee> sortedObservableInterviewees = getBestN(observableInterviewees, comparator,
+                    size);
+            model.setBestNInterviewees(sortedObservableInterviewees);
+            return new CommandResult(MESSAGE_SUCCESS, ToggleView.BEST_INTERVIEWEE);
+
+        } catch (IllegalValueException e) {
+            throw new CommandException(String.format(MESSAGE_PARAM_NOT_FOUND, paramPrefix));
+        }
+    }
+
+    /**
+     *
+     * @param observableInterviewees
+     * @param comparator
+     * @param size
+     * @return
+     */
+    private ObservableList<Interviewee> getBestN(ObservableList<Interviewee> observableInterviewees,
+                                                 Comparator<Interviewee> comparator, int size) {
+        SortedList<Interviewee> sorted = new SortedList<>(observableInterviewees, comparator);
```

Comment:
You need to filter the observableInterviewees for those that have been interviewed first, else you can't compare them by score